synapse/storage/databases/main/purge_events.py

```diff
@@ -368,6 +369,19 @@ def _purge_room_txn(self, txn: LoggingTransaction, room_id: str) -> List[int]:
 
         state_groups = [row[0] for row in txn]
 
+        # Delete tables that don't have an index on `room_id` but do on
+        # `user_id`, so we use `current_state_events` table to find all users
+        # that were in the room and delete those keys.
```

Comment:
Is there a chance of leaving data hanging if there's data from users who aren't in `current_state_events`?

Would it be simpler to add an index on `room_id`? 😄